File: test-serenity2.Web/Modules/ServerTypes/Games/LogsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { LogsRow } from "./LogsRow";

export interface LogsColumns {
    Id: Column<LogsRow>;
    EntityName: Column<LogsRow>;
    EntityId: Column<LogsRow>;
    Action: Column<LogsRow>;
}

export class LogsColumns extends ColumnsBase<LogsRow> {
    static readonly columnsKey = 'Games.Logs';
    static readonly Fields = fieldsProxy<LogsColumns>();
}

File: test-serenity2.Web/Modules/Games/Logs/LogsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace test_serenity2.Games.Columns;

[ColumnsScript("Games.Logs")]
[BasedOnRow(typeof(LogsRow), CheckNames = true)]
public class LogsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string EntityName { get; set; }
    public int EntityId { get; set; }
    public string Action { get; set; }
}